app/services/arcgis_api/geocoder.rb

```diff
@@ -0,0 +1,58 @@
+module ArcgisApi
+  class Geocoder
+    def suggest(text)
+      url = "#{root_url}/suggest"
+      params = {
+        text: text,
+        category: 'address',
+        countryCode: 'USA',
+        f: 'json',
+      }
+
+      parse_suggestions(
+        faraday.get(url, params) do |req|
+          req.options.context = { service_name: 'arcgis_geocoder_suggest' }
+        end.body,
+      )
+    end
+
+    def findAddressCandidates
+    end
+
+    private
+
+    def root_url
+      IdentityConfig.store.arcgis_api_root_url
+    end
+
+    def faraday
+      Faraday.new(headers: request_headers) do |conn|
+        # Log request metrics
+        conn.request :instrumentation, name: 'request_metric.faraday'
+
+        # Raise an error subclassing Faraday::Error on 4xx, 5xx, and malformed responses
+        # Note: The order of this matters for parsing the error response body.
+        conn.response :raise_error
+
+        # Convert body to JSON
+        conn.request :json
+
+        # Parse JSON responses
+        conn.response :json
+      end
+    end
+
+    def request_headers
+      { 'Authorization' => "Bearer #{IdentityConfig.store.arcgis_api_key}" }
+    end
+
+    def parse_suggestions(suggestions)
+      suggestions['suggestions'].map do |suggestion|
+        {
+          text: suggestion['text'],
+          magicKey: suggestion['magicKey'],
+        }
+      end
```

Comment:
just to add on to [this other comment](https://github.com/18F/identity-idp/pull/7186/files#r1001194836) , yes if we're already processing the response, this is a great case for a struct

```suggestion
    Suggestion = Struct.new(:text, :magic_key, keyword_init: true)

    # @return [Array<Suggestion>]
    def parse_suggestions(suggestions)
      suggestions['suggestions'].map do |suggestion|
        Suggestion.new(
          text: suggestion['text'],
          magic_key: suggestion['magic_key'],
        )
      end
```

Reply:
Faraday should automatically throw for the 4xx-coded HTML body. For the 2xx-coded error body, I would suggest specifically checking for that and throwing a custom error. I also recommend a separate error for non-conformant 200 responses.